### PR TITLE
🌱Retrieve kustomize release binary instead of building it

### PR DESCRIPTION
**Pull request by vincepri**
This change came up during #3735, kustomize has been causing issues in
the past as well because it vendors some kubernetes dependencies. To
avoid this problem altogether going forward, let's download the binary
instead of relying on building it locally, or in CI on each run.

/milestone v0.4.0
/assign @fabriziopandini

### CODE REVIEW

File: hack/ensure-kustomize.sh [hunk 1]
```diff
@@ -18,25 +18,38 @@ set -o errexit
...
-      echo 'kustomize not found, installing'
-      if ! [ -d "${GOPATH_BIN}" ]; then
-        mkdir -p "${GOPATH_BIN}"
-      fi
-      curl -sLo "${GOPATH_BIN}/kustomize" https://github.com/kubernetes-sigs/kustomize/releases/download/v${MINIMUM_KUSTOMIZE_VERSION}/kustomize_${MINIMUM_KUSTOMIZE_VERSION}_linux_amd64
-      chmod +x "${GOPATH_BIN}/kustomize"
-    else
-      echo "Missing required binary in path: kustomize"
-      return 2
+  if ! [ -x "$(command -v "${BIN_ROOT}")" ]; then
```
Comment (fabriziopandini) [inline hack/ensure-kustomize.sh:41]: Should this be
```suggestion
  if ! [ -x "$(command -v "${BIN_ROOT}/kustomize")" ]; then
```

**Review by fabriziopandini — COMMENTED** [review]
In the past, I got bitten by differences in kustomize version, but I think this is the less evil when it comes to the transitive dependencies kustomize is bringing in.
Only a small nit, otherwise lgtm

File: hack/ensure-kustomize.sh (hunk 1 above)
Comment (vincepri) [reply hack/ensure-kustomize.sh]: Thanks!

**Comment by vincepri** [comment]
Yeah, this should solve that, so we always use the same version, even locally

**Comment by fabriziopandini** [comment]
/approve
to keep changes fo v1lapha4 moving

**Comment by vincepri** [comment]
FYI, it seems that vendoring kustomize before was causing controller-tools to use older dependencies which is causing now the CRDs to be regenerated, I added a different commit for those

**Pull request merged**